src/Form/matching.js

```diff
@@ -399,6 +405,46 @@ class Matching {
         return undefined
     }
 
+    /**
+     * Returns the password type string including the variant
+     * @param {HTMLInputElement} input
+     * @param opts
+     * @returns {'credentials.password.new'|'credentials.password.current'}
+     */
+    inferPasswordVariant (input, opts) {
+        // Check attributes first
+        // This is done mainly to ensure coverage for all languages, since attributes are usually in English
+        const attrsToCheck = [input.autocomplete, input.name, input.id]
+        if (
+            opts.isSignup &&
+            attrsToCheck.some(str => safeRegexTest(/new.?password|password.?new/i, str))
+        ) {
+            return 'credentials.password.new'
+        }
+        if (
+            (opts.isLogin || opts.isHybrid) &&
+            attrsToCheck.some(str => safeRegexTest(/(current|old|previous).?password|password.?(current|old|previous)/i, str))
+        ) {
+            return 'credentials.password.current'
+        }
```

Comment:
none-blocking question: in your opinion, does this workaround indicate a flaw in our implementation of matchers? 

I only ask because I'm curious if we can expand the design later to include things like this, so that we can keep these regexes together. 

I don't feel strongly, I think your solution here is great, more of a talking point for later :)

Reply:
If I understand your question correctly, unfortunately this has nothing to do with our implementation. It's that some pages use `autocomplete=current-password` for signup pages and `autocomplete=new-password` for login pages, exactly backwards 🤦‍♂️. We found this even on very high-profile pages like Pinterest and Twitch. This workaround is needed to ensure that when that happens we still try to show the best option to users.